### PR TITLE
Sending events to Google Analytics4 as per gtag [TDH-331]

### PR DESCRIPTION
**Pull request by Sanjeevsinghnegi**
### What do you want to achieve?
- Sending events to Google Analytics4 as per gtag

### PR Checklist

- [x] I have tested it locally and all functionalities are working fine.
- [ ] I have compared it with mocks and all design elements are the same.
- [ ] I have tested it in IE Browser.

### How was the code tested?

- By adding ga4 script on local website then intialize the script on that website(the same ga4 id i added on dashboard)

### What new thing you came across while writing this code? 
- ga4 new event firing [instructions](https://developers.google.com/analytics/devguides/collection/ga4/events?client_type=gtag)

### ScreenRecording
- 

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

**Comment by devashishmamgain** [comment]
@Sanjeevsinghnegi are we checking if the customer is using GA4 or older version?

File: webplugin/js/app/km-widget-custom-events.js [hunk 1]
```diff
@@ -12,24 +12,16 @@ var kmWidgetEvents = {
...
+                typeof window.top.gtag !== 'undefined'
             ) {
-                window.top.ga('create', kmWidgetEvents.gaTrackingId(), 'auto');
-                window.top.ga('send', {
-                    hitType: 'event',
-                    eventCategory: eventCategory,
-                    eventAction: eventAction,
-                    eventLabel: eventLabel,
-                    eventValue: eventValue,
+                window.top.gtag('event', eventObject.data.eventAction, {
```
Comment (prochnost04) [inline webplugin/js/app/km-widget-custom-events.js:21]: do we need to update any 3rd party script also?

Comment (Sanjeevsinghnegi) [reply webplugin/js/app/km-widget-custom-events.js:21]: no, basically we are sending events to the user's website ga layer from the widget. there is no direct relation between chat widget and google analytics. Hope this helps. if not let me know

**Comment by Sanjeevsinghnegi** [comment]
> @Sanjeevsinghnegi are we checking if the customer is using GA4 or an older version?

@devashishmamgain 
**Starting July 1, 2023, Universal Analytics properties will stop processing data**

so here we only care about GA4, and we stop supporting UA(Universal analytics) from now as data is not showing there.
If the customer is using UA(older version) then they need to update to ga4

File: webplugin/js/app/km-widget-custom-events.js [hunk 2]
```diff
@@ -80,152 +67,151 @@ var kmWidgetEvents = {
...
+            eventLabel: 'Km Widget CSAT Submit',
         },
         eventFunction: null,
     },
     onShowResolvedClick: {
         data: {
             eventCategory: 'Kommunicate_widget',
             eventAction: 'km_widget_resolve_click',
-            eventLabel: 'Show Resolve',
+            eventLabel: 'Km Widget Show Resolve',
```
Comment (devashishmamgain) [inline webplugin/js/app/km-widget-custom-events.js:126]: from where these names came? what was the issue with earlier label?

**Review by devashishmamgain — CHANGES_REQUESTED** [review]
are we checking if the customer is using older version or newer?

**Review by devashishmamgain — DISMISSED** [review]

File: webplugin/js/app/km-widget-custom-events.js (hunk 2 above)
Comment (Sanjeevsinghnegi) [reply webplugin/js/app/km-widget-custom-events.js]: in ga4 only event action is the only field required so i am sending eventCategory, eventLabel as an additional property under that event action so i added this but it is not req to change the eventLabel so i removed the added prefix

**Review by prochnost04 — DISMISSED** [review]

**Review by panwaranuj01 — APPROVED** [review]

**Review by prochnost04 — APPROVED** [review]

**Review by devashishmamgain — APPROVED** [review]

**Pull request merged**